src/Documentation/tutorials_and_samples/tutorial_1_helloworld.md

```diff
@@ -0,0 +1,151 @@
+        }
+
+```
+
+| Option | Used for |
+|-------------|----------|
+| `.UseLocalhostClustering()` | Same as for SiloHost |
+| `ClusterOptions` | Same as for SiloHost |
+
+A more in-depth guide to configuring your client can be found [in the Client Configuration section of the Configuration Guide.](http://dotnet.github.io/orleans/Documentation/clusters_and_clients/configuration_guide/client_configuration.html)
+
+## Writing a grain
+
+Grains are the key primitives of the Orleans programming model.
+Grains are the building blocks of an Orleans application, they are atomic units of isolation, distribution, and persistence.
+Grains are objects that represent application entities.
+Just like in the classic Object Oriented Programming, a grain encapsulates state of an entity and encodes its behavior in the code logic.
+Grains can hold references to each other and interact by invoking each other’s methods exposed via interfaces.
+
+You can read more about them in the [Core Concepts section of the Orleans documentation.](http://dotnet.github.io/orleans/Documentation/core_concepts/index.html)
+
+This is the main body of code for the Hello World grain:
+
+```csharp
+[...]
+namespace HelloWorld.Grains
+{
+    public class HelloGrain : Orleans.Grain, IHello
+    {
+        Task<string> IHello.SayHello(string greeting)
+        {
+            logger.LogInformation($"SayHello message received: greeting = '{greeting}'");
+            return Task.FromResult($"You said: '{greeting}', I say: Hello!");
+        }
+    }
+}
+```
+
+A grain class implements one or more grain interfaces, as you can read [here, in the Grains section.](http://dotnet.github.io/orleans/Documentation/grains/index.html))
+
+```csharp
+[...]
+namespace HelloWorld.Interfaces
+{
+    public interface IHello : Orleans.IGrainWithIntegerKey
+    {
+        Task<string> SayHello(string greeting);
+    }
+}
+
+```
+
+## How the parts work together
+
+This programming model is built as part of our core concept of distributed Object Oriented Programming.
+SiloHost is run first.
+Then, the OrleansClient program is run.
+The Main method of OrleansClient calls the method that starts the client, `StartClientWithRetries().`
+The client is passed to the `DoClientWork()` method.
+
+
+```csharp
+
+        private static async Task DoClientWork(IClusterClient client)
+        {
+            // example of calling grains from the initialized client
+            var friend = client.GetGrain<IHello>(0);
+            var response = await friend.SayHello("Good morning, my friend!");
+            Console.WriteLine("\n\n{0}\n\n", response);
+        }
+
+```
+
+At this point, OrleansClient creates a reference to the IHello grain and calls its SayHello() method through its interface, IHello.
+Because of this reference, the grain is activated in the silo.
+OrleansClient sends a greeting to the activated grain.
+The grain returns a Task string to OrleansClient, which OrleansClient displays on the console.
```

Comment:
`Task` is not a string, "Task string" is technically incorrect. 

I think "the grain returns the greeting string back to the orleans client, which OrleansClient displays on the console" is good.

Reply:
I wanted to avoid using the word "greeting" here because that's the name of the string being sent to the grain...

```csharp
    public class HelloGrain : Orleans.Grain, IHello
    {
        Task<string> IHello.SayHello(string greeting)
        {
            logger.LogInformation($"SayHello message received: greeting = '{greeting}'");
            return Task.FromResult($"You said: '{greeting}', I say: Hello!");
        }
    ```

Instead, I put the word "response" in there because that's the name of the string being returned and displayed.

Does this seem clearer?

Reply:
So now that whole section looks like this:


At this point, OrleansClient creates a reference to the IHello grain and calls its SayHello() method through its interface, IHello.
This call activates the grain in the silo.
OrleansClient sends a greeting to the activated grain.
The grain returns a response string to OrleansClient, which OrleansClient displays on the console.

Reply:
The line now reads...

The grain returns the greeting as a response to OrleansClient, which OrleansClient displays on the console.

:)